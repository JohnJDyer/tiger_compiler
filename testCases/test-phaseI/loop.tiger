let
  var i : int := 42;
in
  while 1 do
    flush();
    break;
  enddo;
  for i := 0 to 1 do
    break;
    flush();
  enddo;
end
