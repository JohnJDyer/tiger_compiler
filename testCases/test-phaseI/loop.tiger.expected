	assign, i, 42, 
main:
loop_label0:
	breq, 1, 0, loop_label1
	call, flush
	goto, loop_label1, , 
	goto, loop_label0, , 
loop_label1:
	assign, i, 0, 
loop_label2:
	brgt, i, 1, loop_label3
	goto, loop_label3, , 
	call, flush
	add, i, 1, i
	goto, loop_label2, , 
loop_label3:
	return, , , 
